src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -76,6 +76,12 @@ public sealed class LegacyPackageReferenceProject : BuildIntegratedNuGetProject
             return await GetAssetsFilePathAsync(shouldThrow: true);
         }
 
+        public override async Task<string> GetCacheFilePathAsync()
+        {
+            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
We do..the clean cache call is on the UI thread and since we're trying to access the baseintermediateoutputpath we need to be on the UI thread for that.